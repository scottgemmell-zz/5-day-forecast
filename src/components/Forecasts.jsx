import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Day from "./Day";
import spinner from "../assets/svgs/spinner.svg";

export class Forecasts extends Component {

	constructor(props){
		super(props);
		this.filterByDays = this.filterByDays.bind(this);
	}

	filterByDays = data => {
		return (
			data.filter((_k, v) => {
				return (v % 8 === 0) ? true : false;
			})
		);
	};

	render() {
		const { 
			forecasts: { 
				list, 
				city: { name }
			},  
			forecastsLoading,
			forecastsFailure
		} = this.props;
		const filteredList = this.filterByDays(list);

		if (forecastsFailure === true) {
			return <div></div>;
		}

		if (forecastsLoading === true) {
			return <div className="u-spinner"><img src={spinner} alt="Loading..." /></div>;
		}

		return (
			<section>
				<h2 className="section-title">
					Forecasts <em>for</em> {name}
				</h2>

				<div className="forecasts l-panels l-panels@small l-panels@medium l-panels@large">
					{filteredList
						.map((item, i) => {
							return <Day key={i} item={item} />;
						})}
				</div>
			</section>
		)
	}
}

Forecasts.defaultProps = {
	forecasts: {
		list: [],
		city: {
			name: "",
		}
	}
};

Forecasts.propTypes = {
	forecasts: PropTypes.shape({
		list: PropTypes.arrayOf(
			PropTypes.object.isRequired,
		),
		city: PropTypes.shape({
			name: PropTypes.string.isRequired,
		
		}),
	}),
	forecastsLoading: PropTypes.bool,
	forecastsFailure: PropTypes.bool,

};

const mapStateToProps = (state) => ({
	forecasts: state.forecasts.data,
	forecastsFailure: state.forecastsFailure,
	forecastsLoading: state.forecastsLoading
});

export default connect(
	mapStateToProps,
)(Forecasts);