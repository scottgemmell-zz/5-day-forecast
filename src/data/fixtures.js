export const forecasts = {
	city: {
		name: "Aberdeen",
		country: "GB"
	},
	list: [
		{ 
			dt_txt:"2019-01-01 21:00:00",
			main: {
				temp: 4.61
			},
			weather: [
				{
					main: "Clear",
					description: "clear sky"
				}
			]
		}
	]
};