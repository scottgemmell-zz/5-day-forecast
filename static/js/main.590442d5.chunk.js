(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{108:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),c=a(49),l=a.n(c),o=a(10),i=a(13),s=a(51),u=a(52),d=a(53),m=(a(73),a(16)),f=a(17),p=a(20),y=a(18),E=a(19),h=a(110),v=a(118),b=a(11),g="[FORECASTS]",O="".concat(g," FETCH"),_="".concat(g," SET"),j=function(e){var t=e.forecasts;return{type:_,payload:t}},C=a(111),S=a(117),T=function(e){function t(e){var a;return Object(m.a)(this,t),(a=Object(p.a)(this,Object(y.a)(t).call(this,e))).getForecasts=function(e){e.preventDefault(),(0,a.props.fetchForecasts)({city:a.cityInput.value,country:a.countryInput.value})},a.getForecasts=a.getForecasts.bind(Object(b.a)(Object(b.a)(a))),a}return Object(E.a)(t,e),Object(f.a)(t,[{key:"componentDidMount",value:function(){this.cityInput.focus()}},{key:"render",value:function(){var e=this;return r.a.createElement("header",{className:"header"},r.a.createElement(h.a,null,r.a.createElement("h1",{className:"page-title header__title"},"5-day forecast"),r.a.createElement("section",null,r.a.createElement("div",{className:"c-controls"},r.a.createElement(C.a,{inline:!0,onSubmit:this.getForecasts},r.a.createElement("div",{className:"c-controls__inputs"},r.a.createElement(F,{inputRef:function(t){e.cityInput=t},id:"fieldCity",name:"city",placeholderText:"Glasgow",title:"Enter a City",hideLabel:!0,modifier:"c-controls__input c-controls__input--city"})," ",r.a.createElement(F,{inputRef:function(t){e.countryInput=t},id:"fieldCountry",name:"country",placeholderText:"UK",title:"Enter a Country",hideLabel:!0,modifier:"c-controls__input c-controls__input--country"})),r.a.createElement("div",{className:"c-controls__actions"},r.a.createElement(S.a,{bsStyle:"primary",type:"submit"},"Get 5-day forecast")))))))}}]),t}(n.Component),N=Object(o.b)(null,{fetchForecasts:function(e){var t=e.city,a=e.country;return{type:O,payload:{city:t,country:a}}}})(T),w=a(119),I=a(112),x=a(116),R=a(113),D=function(e){var t=e.id,a=e.name,n=e.value,c=e.title,l=e.helpText,o=e.placeholderText,i=e.hideLabel,s=e.modifier,u=e.inputRef;return r.a.createElement(w.a,{controlId:t},r.a.createElement(I.a,{className:i?"u-visibility-hidden":""},c)," ",r.a.createElement(x.a,{inputRef:u,type:"text",name:a,value:n,placeholder:o,bsClass:"form-control ".concat(s)}),r.a.createElement(x.a.Feedback,null),l&&r.a.createElement(R.a,null,l))};D.defaultProps={helpText:"",placeholderText:""};var F=D,k=a(57),A=a.n(k),L=a(114),P=a(58),B=a.n(P),M=function(e){var t=parseFloat(e);return t>8?"hot":t>4?"average":"cold"},U=a(59),Y=a.n(U),G=a(60),H=a.n(G),J=a(61),W=a.n(J),q=a(62),K=a.n(q),Q=a(63),z=a.n(Q),V=a(115),X=function(e){function t(e){var a;return Object(m.a)(this,t),(a=Object(p.a)(this,Object(y.a)(t).call(this,e)))._filterByDays=function(e){return e.filter(function(e,t){return t%8===0})},a.filterByDays=a._filterByDays.bind(Object(b.a)(Object(b.a)(a))),a}return Object(E.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this.props,t=e.forecasts,a=e.forecasts.list,n=e.loading;if(V.a(t))return r.a.createElement("div",null);if(!0===n)return r.a.createElement("div",{className:"u-spinner"},r.a.createElement("img",{src:A.a,alt:"Loading..."}));var c=this._filterByDays(a);return r.a.createElement("section",null,r.a.createElement("h2",{className:"section-title"},"Forecasts ",r.a.createElement("em",null,"for")," ",this.props.forecasts.city.name),r.a.createElement("div",{className:"forecasts l-panels l-panels@small l-panels@medium l-panels@large"},c.map(function(e,t){var a,n=e.dt_txt,c=e.weather,l=e.main.temp;return r.a.createElement("div",{key:t,className:"c-day c-day@small c-day@medium"},r.a.createElement(L.a,{xs:6,className:"c-day__body"},r.a.createElement("dl",null,r.a.createElement("dt",null,"Day:"),r.a.createElement("dd",null,(a=n,B()(a,"YYYY-MM-DD HH:mm:ss").format("dddd"))),r.a.createElement("dt",null,"Weather:"),r.a.createElement("dd",null,"Clear"===c[0].main&&r.a.createElement("div",null,"Clear ",r.a.createElement("br",null),r.a.createElement("small",{className:"notice"},"! No weather icon for clear so using thunder")),"Clear"!==c[0].main&&r.a.createElement("div",null,c[0].main)),r.a.createElement("dt",null,"Description:"),r.a.createElement("dd",null,c[0].description),r.a.createElement("dt",null,"Temperature:"),r.a.createElement("dd",null,r.a.createElement("span",{className:"a-temp a-temp--".concat(M(l))},l,"\xb0")))),r.a.createElement(L.a,{xs:6,className:"c-day__head"},"Rain"===c[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:H.a,alt:"Rain"})),"Clouds"===c[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:Y.a,alt:"Clouds"})),"Clear"===c[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:K.a,alt:"Clear"})),"Sunny"===c[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:W.a,alt:"Sunny"})),"Snowy"===c[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:z.a,alt:"Snowy"}))))})))}}]),t}(n.Component);X.defaultProps={forecasts:{list:"",city:{name:""}}};var Z=Object(o.b)(function(e){return{forecasts:e.forecasts,loading:e.ui.loading}})(X),$=function(e){function t(){return Object(m.a)(this,t),Object(p.a)(this,Object(y.a)(t).apply(this,arguments))}return Object(E.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this.props.notification;return r.a.createElement("div",{className:"app"},r.a.createElement(N,null),r.a.createElement(h.a,null,""===e.message?r.a.createElement(Z,null):r.a.createElement("div",null,r.a.createElement("h2",{className:"section-title"},"Error :("),r.a.createElement(v.a,{bsStyle:"danger",bsClass:"c-controls__alert alert"},"Invalid ",r.a.createElement("strong",null,"City")," and/or ",r.a.createElement("strong",null,"Country"),". [",e.message,"]"))))}}]),t}(n.Component),ee=Object(o.b)(function(e){return{notification:e.notification}})($),te={},ae="API_REQUEST";var ne=function(e){return{type:"SET_LOADER",payload:e.state,meta:{feature:e.feature}}},re=function(e){return{type:"SET_NOTIFICATION",payload:e.message,meta:{status:e.status,feature:e.feature}}};function ce(e){var t=e.city,a=e.country;return"https://api.openweathermap.org/data/2.5/forecast?q=".concat(t,",").concat(a,"&units=metric&appid=e9d4c67ad81c4d73f2ad231f6092f6c3")}var le=a(27),oe={loading:!1},ie={message:""},se=Object(s.createLogger)({collapsed:!0,diff:!0}),ue=Object(u.composeWithDevTools)({}),de=Object(i.combineReducers)({forecasts:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:te,t=arguments.length>1?arguments[1]:void 0;switch(t.type){case _:return t.payload;default:return e}},ui:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:oe,t=arguments.length>1?arguments[1]:void 0;switch(!0){case t.type.includes("SET_LOADER"):return Object(le.a)({},e,{loading:t.payload});default:return e}},notification:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:ie,t=arguments.length>1?arguments[1]:void 0;switch(!0){case t.type.includes("SET_NOTIFICATION"):return Object(le.a)({},e,{message:t.payload});default:return e}}}),me=[function(){return function(e){return function(t){switch(e(t),t.type){case O:e(function(e){var t=e.body,a=e.method,n=e.url,r=e.feature;return{type:"".concat(r," ").concat(ae),payload:t,meta:{method:a,url:n,feature:r}}}({body:null,method:"GET",url:ce({city:t.payload.city,country:t.payload.country}),feature:g})),e(ne({state:!0,feature:g}));break;case"".concat(g," ").concat("API_SUCCESS"):e(j({forecasts:t.payload})),e(re({message:"",feature:g})),e(ne({state:!1,feature:g}));break;case"".concat(g," ").concat("API_ERROR"):e(j({forecasts:t.payload})),e(re({message:t.payload.message,feature:g})),e(ne({state:!1,feature:g}))}}}}],fe=[function(e){var t=e.dispatch;return function(e){return function(a){if(e(a),a.type.includes(ae)){var n=a.meta,r=n.url,c=n.method,l=n.feature;fetch(r,{method:c}).then(function(e){if(e.status>=200&&e.status<300)return e;throw Error(e.statusText)}).then(function(e){return e.json()}).then(function(e){return t(function(e,t){return{type:"".concat(t," ").concat("API_SUCCESS"),payload:e,meta:{feature:t}}}(e,l))}).catch(function(e){return t(function(e){var t=e.error,a=e.feature;return{type:"".concat(a," ").concat("API_ERROR"),payload:t,meta:{feature:a}}}({error:e,feature:l}))})}}}}],pe=Object(i.createStore)(de,ue(i.applyMiddleware.apply(void 0,[d.a].concat(me,fe,[se]))));l.a.render(r.a.createElement(o.a,{store:pe},r.a.createElement(ee,null)),document.getElementById("root"))},57:function(e,t,a){e.exports=a.p+"static/media/spinner.99d7ceaa.svg"},59:function(e,t,a){e.exports=a.p+"static/media/cloudy.ad789581.svg"},60:function(e,t,a){e.exports=a.p+"static/media/rainy-5.2bd23641.svg"},61:function(e,t,a){e.exports=a.p+"static/media/day.7f93fbdb.svg"},62:function(e,t,a){e.exports=a.p+"static/media/thunder.7142c78b.svg"},63:function(e,t,a){e.exports=a.p+"static/media/snowy-6.67d63a14.svg"},64:function(e,t,a){e.exports=a(108)},73:function(e,t,a){}},[[64,1,2]]]);
//# sourceMappingURL=main.590442d5.chunk.js.map