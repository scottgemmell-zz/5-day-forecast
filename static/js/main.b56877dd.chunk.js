(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{108:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),l=a(49),c=a.n(l),i=a(13),s=a(11),o=a(51),u=a(52),d=a(53),m=a(15),h=a(16),y="FORECASTS_FAILURE",f="FORECASTS_LOADING",p=a(54);var E=Object(s.combineReducers)({forecasts:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:p,t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"CITY_UPDATED":case"COUNTRY_UPDATED":return Object(h.a)({},e,Object(m.a)({},t.payload.name,t.payload.value));case"RESET":return Object(h.a)({},e,{city:"",country:"",data:""});case"FORECASTS_SUCCESS":return Object(h.a)({},e,{data:t.forecasts});default:return e}},forecastsFailure:function(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case y:return t.failure;default:return e}},forecastsLoading:function(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case f:return t.loading;default:return e}}}),C=(a(74),a(21)),g=a(22),v=a(25),b=a(23),O=a(26),_=a(9),j=a(111),S=a(117),T=a(112),N=a(115),D=a(116),w=a(109),x=a(114),F=a(110),R=function(e){var t=e.id,a=e.name,n=e.value,l=e.title,c=e.helpText,i=e.placeholderText,s=e.handleChange,o=e.handleBlur,u=e.hideLabel,d=e.modifier;return r.a.createElement(D.a,{controlId:t},r.a.createElement(w.a,{className:u?"u-visibility-hidden":""},l)," ",r.a.createElement(x.a,{type:"text",name:a,value:n,placeholder:i,onChange:s,onBlur:o,bsClass:"form-control ".concat(d)}),r.a.createElement(x.a.Feedback,null),c&&r.a.createElement(F.a,null,c))};R.defaultProps={helpText:"",placeholderText:""};var U=R,B=function(e){var t=e.handleCityBlur,a=e.handleCityChange,n=e.handleCountryBlur,l=e.handleCountryChange,c=e.handleSubmit,i=e.handleReset,s=e.hasErrors;return r.a.createElement("header",{className:"header"},r.a.createElement(j.a,null,r.a.createElement("h1",{className:"page-title header__title"},"5-day forecast"),r.a.createElement("section",null,r.a.createElement("h2",{className:"section-title u-visibility-hidden"},"Controls"),r.a.createElement("div",{className:"c-controls"},r.a.createElement(T.a,{inline:!0,onSubmit:c},r.a.createElement("div",{className:"c-controls__inputs"},r.a.createElement(U,{id:"fieldCity",name:"city",placeholderText:"Glasgow",title:"Enter a City",handleBlur:t,handleChange:a,hideLabel:!0,modifier:"c-controls__input c-controls__input--city"})," ",r.a.createElement(U,{id:"fieldCountry",name:"country",placeholderText:"UK",title:"Enter a Country",handleBlur:n,handleChange:l,hideLabel:!0,modifier:"c-controls__input c-controls__input--country"})," "),r.a.createElement("div",{className:"c-controls__actions"},r.a.createElement(N.a,{type:"reset",onClick:i},"Reset")," ",r.a.createElement(N.a,{bsStyle:"primary",disabled:!s,type:"submit"},"Get 5-day forecast")),!s&&r.a.createElement(S.a,{bsStyle:"info",bsClass:"c-controls__alert alert"},"Please enter a ",r.a.createElement("strong",null,"City")," and a ",r.a.createElement("strong",null,"Country")))))))},L=function(e){return{type:y,failure:e}},k=function(e){return{type:f,loading:e}},A=a(113),Y=a(58),I=a.n(Y),P=function(e){var t=parseFloat(e);return t>8?"hot":t>4?"average":"cold"},G=a(59),M=a.n(G),H=a(60),J=a.n(H),W=a(61),q=a.n(W),K=a(62),z=a.n(K),Q=a(63),V=a.n(Q),X=function(e){var t,a=e.item,n=a.dt_txt,l=a.weather,c=a.main.temp;return r.a.createElement("div",{className:"c-day day@small day@medium"},r.a.createElement(A.a,{xs:6,className:"c-day__body"},r.a.createElement("dl",null,r.a.createElement("dt",null,"Day:"),r.a.createElement("dd",{className:"c-day__dayofweek"},(t=n,I()(t,"YYYY-MM-DD HH:mm:ss").format("dddd"))),r.a.createElement("dt",null,"Weather:"),r.a.createElement("dd",null,"Clear"===l[0].main&&r.a.createElement("div",null,"Clear ",r.a.createElement("br",null),r.a.createElement("small",{className:"notice"},"! No weather icon for clear so using thunder")),"Clear"!==l[0].main&&r.a.createElement("div",null,l[0].main)),r.a.createElement("dt",null,"Description:"),r.a.createElement("dd",null,l[0].description),r.a.createElement("dt",null,"Temperature:"),r.a.createElement("dd",null,r.a.createElement("span",{className:"a-temp a-temp--".concat(P(c))},c,"\xb0")))),r.a.createElement(A.a,{xs:6,className:"c-day__head"},"Rain"===l[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:J.a,alt:"Rain"})),"Clouds"===l[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:M.a,alt:"Clouds"})),"Clear"===l[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:z.a,alt:"Clear"})),"Sunny"===l[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:q.a,alt:"Sunny"})),"Snowy"===l[0].main&&r.a.createElement("div",null,r.a.createElement("img",{src:V.a,alt:"Snowy"}))))},Z=a(64),$=a.n(Z),ee=function(e){function t(e){var a;return Object(C.a)(this,t),(a=Object(v.a)(this,Object(b.a)(t).call(this,e))).filterByDays=function(e){return e.filter(function(e,t){return t%8===0})},a.filterByDays=a.filterByDays.bind(Object(_.a)(Object(_.a)(a))),a}return Object(O.a)(t,e),Object(g.a)(t,[{key:"render",value:function(){var e=this.props,t=e.forecasts,a=t.list,n=t.city.name,l=e.forecastsLoading,c=e.forecastsFailure,i=this.filterByDays(a);return!0===c?r.a.createElement("div",null):!0===l?r.a.createElement("div",{className:"u-spinner"},r.a.createElement("img",{src:$.a,alt:"Loading..."})):r.a.createElement("section",null,r.a.createElement("h2",{className:"section-title"},"Forecasts ",r.a.createElement("em",null,"for")," ",n),r.a.createElement("div",{className:"forecasts l-panels l-panels@small l-panels@medium l-panels@large"},i.map(function(e,t){return r.a.createElement(X,{key:t,item:e})})))}}]),t}(n.Component);ee.defaultProps={forecasts:{list:[],city:{name:""}}};var te=Object(i.b)(function(e){return{forecasts:e.forecasts.data,forecastsFailure:e.forecastsFailure,forecastsLoading:e.forecastsLoading}})(ee),ae=function(e){function t(e){var a;return Object(C.a)(this,t),(a=Object(v.a)(this,Object(b.a)(t).call(this,e))).getForecasts=function(e){e.preventDefault();var t=a.props;(0,t.getForecastData)({city:t.city,country:t.country})},a.handleCityChange=a.handleCityChange.bind(Object(_.a)(Object(_.a)(a))),a.handleCountryChange=a.handleCountryChange.bind(Object(_.a)(Object(_.a)(a))),a.handleReset=a.handleReset.bind(Object(_.a)(Object(_.a)(a))),a}return Object(O.a)(t,e),Object(g.a)(t,[{key:"handleCityChange",value:function(e){var t=this.props.cityUpdated,a=e.target;t({name:a.name,value:a.value})}},{key:"handleCountryChange",value:function(e){var t=this.props.countryUpdated,a=e.target;t({name:a.name,value:a.value})}},{key:"handleReset",value:function(){(0,this.props.reset)()}},{key:"render",value:function(){var e=this.props,t=e.city,a=e.country,n=e.data,l=e.forecastsFailure;return r.a.createElement("div",{className:"app"},r.a.createElement(B,{handleCityChange:this.handleCityChange,handleCountryChange:this.handleCountryChange,handleCityBlur:this.handleCityChange,handleCountryBlur:this.handleCountryChange,handleReset:this.handleReset,handleSubmit:this.getForecasts,hasErrors:t&&a}),r.a.createElement(j.a,null,""!==n&&r.a.createElement(te,null),!0===l&&r.a.createElement("div",null,r.a.createElement("h2",{className:"section-title"},"Error :("),r.a.createElement(S.a,{bsStyle:"danger",bsClass:"c-controls__alert alert"},"Invalid ",r.a.createElement("strong",null,"City")," and/or ",r.a.createElement("strong",null,"Country")))))}}]),t}(n.Component),ne=Object(i.b)(function(e){return{city:e.forecasts.city,country:e.forecasts.country,data:e.forecasts.data,forecastsFailure:e.forecastsFailure,forecastsLoading:e.forecastsLoading}},{cityUpdated:function(e){return{type:"CITY_UPDATED",payload:e}},countryUpdated:function(e){return{type:"COUNTRY_UPDATED",payload:e}},reset:function(){return{type:"RESET"}},getForecastData:function(e){var t=e.city,a=e.country;return function(e){e(k(!0)),fetch("https://api.openweathermap.org/data/2.5/forecast?q=".concat(t,",").concat(a,"&units=metric&appid=").concat("e9d4c67ad81c4d73f2ad231f6092f6c3")).then(function(t){if(!t.ok)throw Error(t.statusText);return e(k(!1)),e(L(!1)),t}).then(function(e){return e.json()}).then(function(t){return e(function(e){return{type:"FORECASTS_SUCCESS",forecasts:e}}(t))}).catch(function(){e(L(!0))})}}})(ae),re=Object(o.createLogger)({collapsed:!0,diff:!0}),le=Object(u.composeWithDevTools)({}),ce=Object(s.createStore)(E,le(Object(s.applyMiddleware)(d.a,re)));c.a.render(r.a.createElement(i.a,{store:ce},r.a.createElement(ne,null)),document.getElementById("root"))},54:function(e){e.exports={city:"",country:"",data:""}},59:function(e,t,a){e.exports=a.p+"static/media/cloudy.ad789581.svg"},60:function(e,t,a){e.exports=a.p+"static/media/rainy-5.2bd23641.svg"},61:function(e,t,a){e.exports=a.p+"static/media/day.7f93fbdb.svg"},62:function(e,t,a){e.exports=a.p+"static/media/thunder.7142c78b.svg"},63:function(e,t,a){e.exports=a.p+"static/media/snowy-6.67d63a14.svg"},64:function(e,t,a){e.exports=a.p+"static/media/spinner.99d7ceaa.svg"},65:function(e,t,a){e.exports=a(108)},74:function(e,t,a){}},[[65,2,1]]]);
//# sourceMappingURL=main.b56877dd.chunk.js.map