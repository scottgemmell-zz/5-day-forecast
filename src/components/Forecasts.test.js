import React from "react";
import { Forecasts } from "./Forecasts";
import { shallow } from "enzyme";
import { forecasts } from "../data/fixtures.js";


const defaultProps = () => ({});
const setup = (overrideProps = {}) => {
	const props = Object.assign({}, defaultProps(), overrideProps);
	const wrapper = shallow(<Forecasts {...props} />);
	return { wrapper, props };
};

describe("<Forecasts />", () => {

	const { wrapper } = setup({ forecasts });

	it("renders", () => {
		expect(wrapper).toMatchSnapshot();
	});

	it("Section Title", () => {
		expect(wrapper.find("section").exists()).toBe(true);
		expect(wrapper.find("h2").exists()).toBe(true);
		expect(wrapper.find("h2").hasClass("section-title")).toBe(true);
		expect(wrapper.find("h2").text()).toBe("Forecasts for Aberdeen");
	});

	//console.log(wrapper.debug());
	
});
