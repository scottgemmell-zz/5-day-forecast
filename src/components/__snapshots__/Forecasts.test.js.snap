// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecasts /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Forecasts
    forecasts={
      Object {
        "city": Object {
          "country": "GB",
          "name": "Aberdeen",
        },
        "list": Array [
          Object {
            "dt_txt": "2019-01-01 21:00:00",
            "main": Object {
              "temp": 4.61,
            },
            "weather": Array [
              Object {
                "description": "clear sky",
                "main": "Clear",
              },
            ],
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="section-title"
        >
          Forecasts 
          <em>
            for
          </em>
           
          Aberdeen
        </h2>,
        <div
          className="forecasts l-panels l-panels@small l-panels@medium l-panels@large"
        >
          <Day
            item={
              Object {
                "dt_txt": "2019-01-01 21:00:00",
                "main": Object {
                  "temp": 4.61,
                },
                "weather": Array [
                  Object {
                    "description": "clear sky",
                    "main": "Clear",
                  },
                ],
              }
            }
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Forecasts ",
            <em>
              for
            </em>,
            " ",
            "Aberdeen",
          ],
          "className": "section-title",
        },
        "ref": null,
        "rendered": Array [
          "Forecasts ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "for",
            },
            "ref": null,
            "rendered": "for",
            "type": "em",
          },
          " ",
          "Aberdeen",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Day
              item={
                Object {
                  "dt_txt": "2019-01-01 21:00:00",
                  "main": Object {
                    "temp": 4.61,
                  },
                  "weather": Array [
                    Object {
                      "description": "clear sky",
                      "main": "Clear",
                    },
                  ],
                }
              }
            />,
          ],
          "className": "forecasts l-panels l-panels@small l-panels@medium l-panels@large",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "item": Object {
                "dt_txt": "2019-01-01 21:00:00",
                "main": Object {
                  "temp": 4.61,
                },
                "weather": Array [
                  Object {
                    "description": "clear sky",
                    "main": "Clear",
                  },
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="section-title"
          >
            Forecasts 
            <em>
              for
            </em>
             
            Aberdeen
          </h2>,
          <div
            className="forecasts l-panels l-panels@small l-panels@medium l-panels@large"
          >
            <Day
              item={
                Object {
                  "dt_txt": "2019-01-01 21:00:00",
                  "main": Object {
                    "temp": 4.61,
                  },
                  "weather": Array [
                    Object {
                      "description": "clear sky",
                      "main": "Clear",
                    },
                  ],
                }
              }
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Forecasts ",
              <em>
                for
              </em>,
              " ",
              "Aberdeen",
            ],
            "className": "section-title",
          },
          "ref": null,
          "rendered": Array [
            "Forecasts ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "for",
              },
              "ref": null,
              "rendered": "for",
              "type": "em",
            },
            " ",
            "Aberdeen",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Day
                item={
                  Object {
                    "dt_txt": "2019-01-01 21:00:00",
                    "main": Object {
                      "temp": 4.61,
                    },
                    "weather": Array [
                      Object {
                        "description": "clear sky",
                        "main": "Clear",
                      },
                    ],
                  }
                }
              />,
            ],
            "className": "forecasts l-panels l-panels@small l-panels@medium l-panels@large",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "item": Object {
                  "dt_txt": "2019-01-01 21:00:00",
                  "main": Object {
                    "temp": 4.61,
                  },
                  "weather": Array [
                    Object {
                      "description": "clear sky",
                      "main": "Clear",
                    },
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
